token-service/src/main/java/org/p2p/token/service/repository/TokenServiceRemoteRepository.kt

```diff
@@ -0,0 +1,64 @@
+package org.p2p.token.service.repository
+
+import org.p2p.token.service.api.TokenServiceApiRepository
+import org.p2p.token.service.api.mapper.TokenServiceMapper
+import org.p2p.token.service.api.request.TokenServiceItemRequest
+import org.p2p.token.service.api.request.TokenServiceMetadataRequest
+import org.p2p.token.service.api.request.TokenServicePriceRequest
+import org.p2p.token.service.api.request.TokenServiceQueryRequest
+import org.p2p.token.service.model.TokenServiceMetadata
+import org.p2p.token.service.model.TokenServiceNetwork
+import org.p2p.token.service.model.TokenServicePrice
+import org.p2p.token.service.model.TokenServiceQueryResult
+import org.p2p.token.service.model.unwrap
+
+class TokenServiceRemoteRepository(
+    private val api: TokenServiceApiRepository,
+    private val mapper: TokenServiceMapper
+) : TokenServiceRepository {
+
+    override suspend fun loadTokensPrice(
+        chain: TokenServiceNetwork,
+        addresses: List<String>
+    ): List<TokenServiceQueryResult<TokenServicePrice>> {
+
+        val queryRequest = createRequest(chain, addresses).let(::TokenServicePriceRequest)
+        val queryResponse = api.launch(queryRequest).unwrap().orEmpty()
```

Comment:
.unwrap() and Result class in the end is not used at all
Result should be used to explicitly show what this method returns and instead of that you silently throw exception inside